### DISCUSSION TITLE
Tags are not being sent through SentrySdk.CaptureException? .NET 7 API

### DISCUSSION THREAD

**Discussion by platform-co** [getsentry/sentry-dotnet#2497] [Q&A] [answered]
Hello,

I have an .net 7 api with the follwoing sentry setup:
`    builder.WebHost.UseSentry(options =>
    {
        options.DefaultTags.Add("Customer", customer.ToString());
    });`

This default tag is show in the event. But when I add other tags:

`SentrySdk.CaptureException(error.Exception, scope =>
                {
                    scope.SetTag("ErrorType", nameof(ResolverError));
                    scope.SetTag("transaction_id", transactionId);
                    scope.SetFingerprint(error.Code ?? "Generic", error.Message);
                });`

These Tags arn't visbile. Not the ErrorType nor the transaction_id.

There are also no later scopes being set or overwriten. But I would also think that this should not be posible as the scope is incapsulated in the CaptureException. And should not have effect on other events nor should other events have effect on this event.

I also tried other methods like  SentrySdk.ConfigureScope with SentrySdk.PushScope() or the older SentrySdk.WithScope( but nothing seems to solve my problem.

Any ideas what I might be missing here?

**Comment by platform-co** [comment]
I created a new .NET 7 WebAPI added:

`builder.WebHost.UseSentry(options =>
{
    options.DefaultTags.Add("DefaultTag", "Initial");
});`

and:

`SentrySdk.CaptureException(new Exception(), scope =>
        {
            scope.SetTag("AditionalTag", "AnotherTag");
        });`

And this does work. Checking out the differences and will update the findings here.

**Comment by platform-co** [comment]
The capture exception was never entering sentry. I was doing a log entry right before it:
`_logger.LogError(error.Exception, "A resolver error occurred: [{Code}] {Message}", error.Code, error.Message);`

This somehow prevented the captureexception from ending up in sentry:
`SentrySdk.CaptureException(error.Exception, scope => { scope.SetTag("ErrorType", nameof(ResolverError)); scope.SetTag("transaction_id", transactionId); scope.SetFingerprint(error.Code ?? "Generic", error.Message); });`

Not sure yet why this is the case?

**Comment by jamescrosswell** [comment, marked as answer]
Hey @platform-co ,

If you configure Sentry with the Debug option I think you'll see a Debug log message explaining what's going. 
```csharp
builder.WebHost.UseSentry(options => { 
  // ... other options
  options.DefaultTags.Add("DefaultTag", "Initial"); 
  options.Debug = true; 
});
```

I think you'll see something like the following in your Debug Output when you do this:

```
Event dropped by processor DuplicateEventDetectionEventProcessor
```

Essentially the exception is being sent through to sentry twice. Once when you call `_logger.LogError` and once when you call `SentrySdk.CaptureException`... so sentry drops the second one. 

One solution is that you don't pass the exception as a first parameter to `_logger.LogError` - so just supply the text that you want logged:
```csharp
_logger.LogError("A resolver error occurred: [{Code}] {Message}", error.Code, error.Message);
```

In Sentry, the `LogError` message will show up as a breadcrumb in the details for the Issue/Exception so you'll have everything in one place anyway (Exception details, stack track, default tags, custom tags and the log error message).

**Reply by platform-co** [reply]
Hey James,

Thanks for clearing this up. I ended up removing the log message.
